src/sparseml/transformers/sparsification/sparse_model.py

```diff
@@ -131,11 +132,13 @@ def skip(*args, **kwargs):
 
         recipe = resolve_recipe(recipe=recipe, model_path=pretrained_model_name_or_path)
         if recipe:
-            apply_recipe_structure_to_model(
-                model=model,
-                model_path=pretrained_model_name_or_path,
-                recipe_path=recipe,
-            )
+            with session_context_manager():
```

Comment:
It doesn't seem like this context manager will create a session if it doesn't already exist since it just queries the active session, how does this work?

Reply:
https://github.com/neuralmagic/sparseml/blob/fd0a7793c2a1b27d7300f40a5f8d1e9a58f9a34d/src/sparseml/core/session.py#L388 if the active session is not present it defaults to the global_session in the linked file so we should be fine in that respect.